### PR TITLE
Use upstream go-yaml fork and remove our internal one

### PR DESCRIPTION
**Pull request by natasha41575**
https://github.com/kubernetes-sigs/yaml/pull/76 is merged, so we can get rid of our internal fork of go-yaml and use the upstream one instead.

### CODE REVIEW

File: go.work.sum [hunk 1]
```diff
@@ -145,6 +145,7 @@ github.com/moby/spdystream v0.2.0 h1:cjW1zVyyoiM0T7b6UoySUFqzXMoqRckQtXwGPiBhOM8
 github.com/moby/spdystream v0.2.0/go.mod h1:f7i0iNDQJ059oMTcWxx8MA/zKFIuD/lY+0GqbN2Wy8c=
 github.com/mwitkow/go-conntrack v0.0.0-20161129095857-cc309e4a2223 h1:F9x/1yl3T2AeKLr2AMdilSD8+f9bvMnNN8VS5iDtovc=
 github.com/mxk/go-flowrate v0.0.0-20140419014527-cca7078d478f h1:y5//uYreIhSUg3J1GEMiLbxo1LJaP8RfCpH6pymGZus=
+github.com/natasha41575/yaml-1 v1.3.1-0.20231023160344-f2c02299d455/go.mod h1:Ejl7/uTz7PSA4eKMyQCUTnhZYNmLIl+5c2lQPGR2BPY=
```
Comment (ncapps) [inline go.work.sum:148]: Is this meant to point to your fork of yaml?

Comment (natasha41575) [reply go.work.sum:148]: Oh my bad, no it isn't. Let me try to fix that, thanks!

Comment (natasha41575) [reply go.work.sum]: Fixed

**Review by ncapps — COMMENTED** [review]
I'm not technically a reviewer but this LGTM.

**Comment by rayandas** [comment]
Went through the changes. It looks good. 
/lgtm

**Comment by rayandas** [comment]
Why did it merge the PR (with lgtm from a non-approver) ?? 😢

**Pull request merged**